src/orion/benchmark/assessment/base.py

```diff
@@ -25,24 +25,38 @@ class BenchmarkAssessment(ABC):
     def __init__(self, task_num, **kwargs):
         self.task_number = task_num
         self._param_names = kwargs
-        self._param_names["task_num"] = task_num
+
+        self._param_names["task_num"] = (
+            int(task_num / len(kwargs.get("n_workers")))
+            if kwargs.get("n_workers")
+            else task_num
+        )
 
     @property
     def task_num(self):
         """Return the task number to run for this assessment"""
         return self.task_number
 
+    def executor(self, task_index):
+        """Return an instance of `orion.executor.base.Executor` based on the index of tasks
+        that the assessment is asking to run."""
+        return None
+
     @abstractmethod
     def analysis(self, task, experiments):
         """
-        Generate a `plotly.graph_objects.Figure` to display the performance analysis
+        Generate `plotly.graph_objects.Figure` objects to display the performance analysis
         based on the assessment purpose.
 
         task: str
             Name of the task
         experiments: list
             A list of (task_index, experiment), where task_index is the index of task to run for
             this assessment, and experiment is an instance of `orion.core.worker.experiment`.
+
+        Returns
+        -------
+        Dict of plotly.graph_objects.Figure objects
```

Comment:
We should probably clarify the structure of dict we expect. Would it be `[assessment_name][task_name]`?